demos/spawn/README.md

```diff
@@ -0,0 +1,40 @@
+# Spawn
+
+In this demo we are going to show the mechanism that a thread can spawn other threads.
+
+To illustrate this demo, we need to introduce the following:
+
+* `add_named_thread` (in `src/user/thread.c`)
+* `thread_name` (in `src/user/thread.c`)
+
+## `add_named_thread`
+
+`add_named_thread` uses the `name` and `worker` we provided to create a thread and initializes all its arguments to `0`.
+
+If the creation successes, the function returns the index of the thread as the return value. If the creation fails, the return value will be set to `INVALID_THREAD`.
+
+After the creation, the thread starts to wait until it is yielded to.
+
+## `thread_name`
+
+In AMT, we store the name of a thread in a `Thread` structure. 
+
+When we call this function, it uses the `get_thread_property` syscall to copy the name from the kernel, then stores it into the location pointed to by `name`. 
+
+## Walkthrough
+
+We start from the `setup()` function. The `setup()` function is just a starting point for thread 0 to run, so we directly jump into `spawner()` function.
+
+When we enter the function, it first sets the thread name to `spawner`, then creates 3 threads called `Morgoth`, `Sauron` and `Khamul`. After each thread has been created, we send a message to it. These messages indicate how many turns it should wait to log its greeting.
```

Comment:
If you were wondering about the names, I just googled for obscure Lord of the Rings character names :)